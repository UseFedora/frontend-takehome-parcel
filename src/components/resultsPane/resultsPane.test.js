import React from 'react';
import { shallow } from 'enzyme';
import { DotLoader } from 'react-spinners';
import ResultsPane from './';

describe('ResultsPane', () => {
  it('should render correctly', () => {
    const component = shallow(<ResultsPane />);
    expect(component).toMatchSnapshot();
  });

  it('should render the spinner when loading', () => {
    const component = shallow(
      <ResultsPane
        isLoading={true}
      />
    );
    expect(component.find(DotLoader).exists()).toBeTruthy();
  });

  it('should display the empty results message', () => {
    const component = shallow(
      <ResultsPane
        numberOfResults={0}
        emptyResultsMessage="Nothing going on here"
      />
    );
    expect(component.text()).toContain('Nothing going on here');
  });

  it('should not display the empty results message if there are results', () => {
    const component = shallow(
      <ResultsPane
        numberOfResults={1}
        emptyResultsMessage="Nothing going on here"
      />
    );
    expect(component.text()).not.toContain('Nothing going on here');
  });

  it('should render the supplied component', () => {
    const component = shallow(
      <ResultsPane
        numberOfResults={1}
      >
        <ul className="a-result"></ul>
      </ResultsPane>
    );
    expect(component.find('ul.a-result').exists()).toBeTruthy();
  });
});
