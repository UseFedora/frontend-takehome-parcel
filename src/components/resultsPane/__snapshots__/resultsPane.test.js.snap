// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPane should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsPane
    emptyResultsMessage=""
    isLoading={false}
    numberOfResults={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <p>
          
        </p>
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          
        </p>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": undefined,
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <p>
            
          </p>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            
          </p>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
